cassandra/src/main/scala/filodb.cassandra/columnstore/IngestionTimeIndexTable.scala

```diff
@@ -41,37 +36,40 @@ sealed class IngestionTimeIndexTable(val dataset: DatasetRef, val connector: Fil
                      |) WITH compression = {
                     'sstable_compression': '$sstableCompression'}""".stripMargin
 
-  def fromRow(row: Row): InfoRecord =
-    InfoRecord(row.getBytes("partition"), row.getBytes("info"))
-
-  val selectCql = s"SELECT partition, info FROM $tableString WHERE "
-  lazy val allPartReadCql = session.prepare(selectCql + "partition = ?")
-  lazy val inPartReadCql = session.prepare(selectCql + "partition IN ?")
+  lazy val allCql = session.prepare(
+      s"SELECT ingestion_time, start_time, info FROM $tableString " +
+      s" WHERE partition = ?")
+    .setConsistencyLevel(ConsistencyLevel.ONE)
 
   /**
-   * Retrieves all infos from a single partition.
-   */
-  def getInfos(binPartition: Array[Byte]): Observable[InfoRecord] = {
-    val it = session.execute(allPartReadCql.bind(toBuffer(binPartition)))
-                    .asScala.toIterator.map(fromRow)
-    Observable.fromIterator(it).handleObservableErrors
-  }
-
-  def getMultiInfos(partitions: Seq[Array[Byte]]): Observable[InfoRecord] = {
-    val query = inPartReadCql.bind().setList(0, partitions.map(toBuffer).asJava, classOf[ByteBuffer])
-    val it = session.execute(query).asScala.toIterator.map(fromRow)
-    Observable.fromIterator(it).handleObservableErrors
+    * Test method which returns all rows for a partition.
+    */
+  def readAllRows(partKeyBytes: ByteBuffer): ResultSet = {
```

Comment:
Should explicitly call out APIs that are not async and not usable in real-time server.
If used, they can potentially block ingestion or query threads and slow down the server.

Reply:
Good point.